[WPE] Support the stable xdg_shell protocol in WindowViewBackend

zdobersek opened the pull request:
#### 2dc61a55832a0a6e75c963cd3805ab645c5dad7c
<pre>
[WPE] Support the stable xdg_shell protocol in WindowViewBackend
<a href="https://bugs.webkit.org/show_bug.cgi?id=247176">https://bugs.webkit.org/show_bug.cgi?id=247176</a>

Reviewed by Carlos Garcia Campos.

Have WPE&apos;s WindowViewBackend support the stable xdg_shell protocol, implementing
it alongside the support for the unstable variant of that protocol. The stable
variant is preferred, if available.

The relevant protocol files are generated and included in the build. The shell
objects and listeners are reorganized into two separate structs. The differences
between the two variants are not big, main problem is just the different set
of types and functions being used.

* Tools/wpe/backends/PlatformWPE.cmake:
* Tools/wpe/backends/fdo/WindowViewBackend.cpp:
(WPEToolingBackends::WindowViewBackend::onDOMFullscreenRequest):
(WPEToolingBackends::WindowViewBackend::WindowViewBackend):
(WPEToolingBackends::WindowViewBackend::~WindowViewBackend):
* Tools/wpe/backends/fdo/WindowViewBackend.h:

Canonical link: <a href="https://commits.webkit.org/256092@main">https://commits.webkit.org/256092@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f381731e7d02a1361c0f93e69882b8869994c602

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/94657 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/3816 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/27538 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/104279 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/164548 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/98653 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/3882 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/32007 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/86963 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/100232 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/100324 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/2790 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/81023 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/29820 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/84716 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/84274 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/72708 "Found 2 new API test failures: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/state-changed, /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/event-listener (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/38411 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/18090 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/36262 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/19367 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4206 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/40169 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/42016 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/42138 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/38600 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Tools/wpe/backends/fdo/WindowViewBackend.cpp:
```diff
@@ -516,15 +527,76 @@ const struct wl_seat_listener WindowViewBackend::s_seatListener = {
+            uint32_t state = *pos;
+
+            switch (state) {
+            case ZXDG_TOPLEVEL_V6_STATE_ACTIVATED:
```
I think this is `XDG_TOPLEVEL_STATE_ACTIVATED` in the stable protocol

zdobersek replied:
Yeah, copied too eagerly.

carlosgcampos commented on Tools/wpe/backends/fdo/WindowViewBackend.cpp:
```diff
@@ -516,15 +527,76 @@ const struct wl_seat_listener WindowViewBackend::s_seatListener = {
+                isFullscreen = true;
+                break;
+            case ZXDG_TOPLEVEL_V6_STATE_MAXIMIZED:
+            case ZXDG_TOPLEVEL_V6_STATE_RESIZING:
```
And the same for the other states.

zdobersek committed:
[WPE] Support the stable xdg_shell protocol in WindowViewBackend
https://bugs.webkit.org/show_bug.cgi?id=247176

Reviewed by Carlos Garcia Campos.

Have WPE's WindowViewBackend support the stable xdg_shell protocol, implementing
it alongside the support for the unstable variant of that protocol. The stable
variant is preferred, if available.

The relevant protocol files are generated and included in the build. The shell
objects and listeners are reorganized into two separate structs. The differences
between the two variants are not big, main problem is just the different set
of types and functions being used.

* Tools/wpe/backends/PlatformWPE.cmake:
* Tools/wpe/backends/fdo/WindowViewBackend.cpp:
(WPEToolingBackends::WindowViewBackend::onDOMFullscreenRequest):
(WPEToolingBackends::WindowViewBackend::WindowViewBackend):
(WPEToolingBackends::WindowViewBackend::~WindowViewBackend):
* Tools/wpe/backends/fdo/WindowViewBackend.h:

Canonical link: https://commits.webkit.org/256092@main
